xcm/update_to_prod.py

```diff
@@ -0,0 +1,134 @@
+
+    return prod
+
+
+def update_destinations(dev, prod, meta_dict):
+    for chain_id, chain in prod.items():
+        for asset_location, asset in chain['assets'].items():
+            for destination_id, destination in dev[chain_id]['assets'][asset_location]['xcmTransfers'].items():
+                if destination_id not in asset['xcmTransfers']:
+                    print(f"Was added new destination in {meta_dict[chain_id]['name']} \
+                          \nfor asset: {asset_location} \
+                          \nto network: {meta_dict[destination_id]['name']} ")
+                    if ask_to_update():
+                        prod[chain_id]['assets'][asset_location]['xcmTransfers'][destination_id] = destination
+
+    return prod
+
+
+def convert_chains_to_dict(chains_obj):
+    chains_dict = {chain['chainId']: chain for chain in chains_obj}
+    for chain_id, chain in chains_dict.items():
+        assets_dict = {}
+        for asset in chain['assets']:
+            destinations_dict = {destination['destination']['chainId']: destination for destination in asset['xcmTransfers']}
+            assets_dict[asset['assetLocation']] = asset
+            assets_dict[asset['assetLocation']]['xcmTransfers'] = destinations_dict
+        chains_dict[chain_id]['assets'] = assets_dict
+
+    return chains_dict
+
+def convert_chain_dict_to_array_back(chains_dict):
+    temp_chains = []
+    for _, chain in chains_dict.items():
+        temp_assets = []
+        for _, asset in chain['assets'].items():
+            temp_destinations = []
+            for _, destination in asset['xcmTransfers'].items():
+                temp_destinations.append(destination)
+            asset['xcmTransfers'] = temp_destinations
+            temp_assets.append(asset)
+        chain['assets'] = temp_assets
+        temp_chains.append(chain)
+
+    return temp_chains
+
+
+
+def promote_updates_to_prod(dev_file, prod_file, meta_data):
+    meta_dict = {chain['chainId']: chain for chain in meta_data}
+
+    dev_chains_dict = convert_chains_to_dict(dev_file['chains'])
+    prod_chains_data = convert_chains_to_dict(prod_file['chains'])
+
+    updated_prod = update_networks(dev_chains_dict, prod_chains_data, meta_dict)
+    updated_prod = update_assets(dev_chains_dict, updated_prod, meta_dict)
+    updated_prod = update_destinations(dev_chains_dict, updated_prod, meta_dict)
+
+    updated_chains = convert_chain_dict_to_array_back(updated_prod)
+
+    updated_base_weight = update_network_base_weight(dev_file, prod_file, meta_dict)
+
+    updated_reserves = update_reserves(dev_file, prod_file)
+
+    prod_file['networkBaseWeight'] = updated_base_weight
+    prod_file['assetsLocation'] = updated_reserves
+    prod_file['chains'] = updated_chains
+
+    return prod_file
+
+
+if __name__ == "__main__":
+    # Read data from files
+    with open('xcm/v2/transfers_dev.json') as f:
+        data1 = json.load(f)
+    with open('xcm/v2/transfers.json') as f:
```

Comment:
(In general applies to all scripsts)

I suggest not to hardcode version (v2) in each script since it makes changing the version a little bit painful. Instead, we can store it somewhere globally, or even better, detect automatically be check the contents of xcm/ folder

Reply:
Added getting all paths from environment variables. Basically, we have workflow to setting the las env variables - https://github.com/nova-wallet/nova-utils/blob/f6ed2e81e8fc71d34a23edacd02632c5a97b74db/.github/workflows/setup-path/action.yml#L8